tests/momento/auth/test_credential_provider.py

```diff
@@ -45,6 +53,44 @@
             "give.me.control.test.com",
             "secret.cache.test.com",
         ),
+        (
+            CredentialProvider.from_string(auth_token=test_encoded_token),
+            test_v1_api_key,
+            "control.test.momentohq.com",
+            "cache.test.momentohq.com",
+        ),
+        (
+            CredentialProvider.from_string(auth_token=test_encoded_token.decode("utf-8")),
+            test_v1_api_key,
+            "control.test.momentohq.com",
+            "cache.test.momentohq.com",
+        ),
+        (
+            CredentialProvider.from_string(
+                auth_token=test_encoded_token,
+                control_endpoint="give.me.control.test.com",
+                cache_endpoint="secret.cache.test.com",
+            ),
+            test_v1_api_key,
+            "give.me.control.test.com",
+            "secret.cache.test.com",
+        ),
+        (
+            CredentialProvider.from_environment_variable(env_var_name=test_v1_env_var_name),
+            test_v1_api_key,
+            "control.test.momentohq.com",
+            "cache.test.momentohq.com",
+        ),
+        (
+            CredentialProvider.from_environment_variable(
+                env_var_name=test_v1_env_var_name,
+                control_endpoint="give.me.control.test.com",
+                cache_endpoint="secret.cache.test.com",
+            ),
+            test_v1_api_key,
+            "give.me.control.test.com",
+            "secret.cache.test.com",
+        ),
```

Comment:
does this cover all the test cases that @bruuuuuuuce added in his second PR against the node.js SDK?

Reply:
I believe so-- v1 vs legacy crossed with string vs env var crossed with endpoints overridden vs endpoints not overridden. @bruuuuuuuce please let me know if you see anything I'm missing here.

Reply:
My only question is about this test 

```
CredentialProvider.from_string(auth_token=test_encoded_v1_token.decode("utf-8"))
```

Are we supporting decoding the tokens and passing in the decoded stringified json value to the credentials provider? If so I don't think we support this in nodejs yet

Reply:
Talked to Matt offline about this and we're all good. The conversion there is just for mypy's benefit.